runtime/src/nft/mod.rs

```diff
@@ -0,0 +1,169 @@
+        AccountForAsset get(fn account_for_asset): double_map hasher(blake2_128_concat) RegistryId, hasher(blake2_128_concat) TokenId => T::AccountId;
+        /// A double mapping of registry id and asset id to an asset's info.
+        Assets get(fn asset): double_map hasher(blake2_128_concat) RegistryId, hasher(blake2_128_concat) TokenId => <T as Trait>::AssetInfo;
+    }
+}
+
+// Empty event to satisfy type constraints
+decl_event!(
+    pub enum Event<T> where
+        <T as frame_system::Trait>::AccountId,
+    {
+        /// Ownership of the asset has been transferred to the account.
+        Transferred(RegistryId, AssetId, AccountId),
+    }
+);
+
+decl_error! {
+    pub enum Error for Module<T: Trait> {
+        // Thrown when there is an attempt to mint a duplicate asset.
+        AssetExists,
+        // Thrown when there is an attempt to transfer a nonexistent asset.
+        NonexistentAsset,
+        // Thrown when someone who is not the owner of a asset attempts to transfer or burn it.
+        NotAssetOwner,
+    }
+}
+
+// Empty module so that storage can be declared
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+        type Error = Error<T>;
+        fn deposit_event() = default;
+
+        /// Transfer a asset to a new owner.
+        ///
+        /// The dispatch origin for this call must be the asset owner.
+        ///
+        /// - `dest_account`: Receiver of the asset.
+        /// - `asset_id`: The hash (calculated by the runtime system's hashing algorithm)
+        ///   of the info that defines the asset to destroy.
+        #[weight = T::DbWeight::get().reads_writes(1,1) + 195_000_000]
+        pub fn transfer(origin,
+                        dest_account: T::AccountId,
+                        registry_id: RegistryId,
+                        token_id: TokenId)
+        -> dispatch::DispatchResult {
+            let who = ensure_signed(origin)?;
+
+            let asset_id = AssetId(registry_id, token_id);
+            <Self as Unique>::transfer(&who, &dest_account, &asset_id)?;
+
+            Self::deposit_event(RawEvent::Transferred(registry_id, asset_id, dest_account));
+
+            Ok(())
+        }
+    }
+}
+
+impl<T: Trait>
+    Unique for Module<T>
+{
+    type Asset = Asset<AssetId, <T as Trait>::AssetInfo>;
+    type AccountId = <T as frame_system::Trait>::AccountId;
+
+    fn owner_of(asset_id: &AssetId) -> T::AccountId {
+        let (registry_id, token_id) = AssetIdRef::from(asset_id).destruct();
+        Self::account_for_asset(registry_id, token_id)
+    }
+
+    fn transfer(
+        caller: &T::AccountId,
+        dest_account: &T::AccountId,
+        asset_id: &AssetId,
+    ) -> dispatch::DispatchResult {
+        let owner = Self::owner_of(asset_id);
+        let (registry_id, token_id) = AssetIdRef::from(asset_id).destruct();
+
+        // Check that owner account exists
+        ensure!(owner != T::AccountId::default(),
```

Comment:
Does this check if the destination account exists as of has the existential deposit?

Reply:
`AccountId::default()` is the return value if no owner exists. The approach is confusing so I switched to using an Option type check and removed this line.